TentacleSoftware.Telnet/TelnetClient.cs

```diff
@@ -87,7 +87,7 @@ public async Task Connect(string socks4ProxyHost, int socks4ProxyPort, string so
 
             // Simple implementation of http://en.wikipedia.org/wiki/SOCKS#SOCKS4
             // Similar to http://biko.codeplex.com/
-            byte[] hostAddress = Dns.GetHostAddresses(_host).First().GetAddressBytes();
+            byte[] hostAddress = Dns.GetHostAddresses(_host).First(addr => addr.GetAddressBytes().Length == 4).GetAddressBytes();
```

Comment:
Would `a => a.AddressFamily == AddressFamily.InterNetwork` work as the selector?